README.md

```diff
@@ -144,32 +156,36 @@ traffic_profile: <TRAFFIC_PROFILE>
 private_base_url: <PRIVATE_BASE_URL>
 ```
 
+*Installation Step*:
+
 **3.** Execute the playbook `profiled-injector.yml` to deploy injector.
 
 ```bash
 ansible-playbook -i hosts profiled-injector.yml
+
 ```
 
-## Run tests
+## Configure test run
 
-**1.** Configure performance parameters:
+**1.** Configure users per run-phase parameters:
 
 ```
-File: run.yml
+File: group_vars/all.yml
 
-<RPS>: Maximum requests per second to send
-<DURATION>: Duration of the performance test in seconds
-<THREADS>: Number of threads (parallel connections) to use
+ramp_up_target_users_per_sec: 1
```

Comment:
Can you try to keep the same interface
```
<RPS>: Maximum requests per second to send
<DURATION>: Duration of the performance test in seconds
<THREADS>: Number of threads (parallel connections) to use
```

and do all the settings of hyperfoil behind the scenes?

Reply:
Yes I can reuse the same interface.
<RPS>: this gets mapped to ramp_up_target_users_per_sec, steady_state_users and ramp_down_users
<DURATION>: this gets mapped to steady_state_duration
<THREADS>: this gets mapped to injector_hyperfoil_agent_threads
 Adding these new variables in to the next commit.

Reply:
There is not a good match. It might be my fault with names. Let's clarify.

RPS is the request per second. It is not the number of users (http concurrent connections). The interface states a max requests per second generated by the client. You need to do a throughput shaping to ensure you do not create more requests than required.
THREADS is the number of TCP concurrent connections where HTTP session are being held. What is exactly `injector_hyperfoil_agent_threads`?

Reply:
waiting for feedback, as this is the main issue to accept an injector update

Reply:
RPS, in this situation with [Hyperfoil](https://hyperfoil.io/docs/intro.html) the model is not what you would normally understand with JMeter. With Hyperfoil the parameters `ramp_up_target_users_per_sec` , `steady_state_users` and `ramp_down_users` are "user requests per second". Not Users. Also Hyperfoil automatically limits the requests. Avoiding the need for an explicit traffic shaping handler which is necessary with JMeter.

 Active connections (per target host) are configured using the property `shared_connections` found [here](https://github.com/whitingjr/perftest-toolkit/blob/4925a13bdf77690998aa2bef437af538f2df58de/deployment/group_vars/all.yml#L7).

 THREADS, with Hyperfoil the threads do not equate to TCP concurrent connections. You need to leave that understanding of the model behind with JMeter. Instead THREADS is the number of Worker Threads that process blocking execution.
 To configure the Worker Thread pool the property `injector_hyperfoil_agent_threads` is used. It's the closest concept of Threads you'll find in Hyperfoil.

 Hyperfoil has been developed using reactive style based on an [event loop](https://en.wikipedia.org/wiki/Event_loop). Which means the [Open System Model](https://hyperfoil.io/docs/intro.html#accuracy) of modelling a workload creates equivalent requests using fewer resources.